pkg/agent/multicast/mcast_discovery.go

```diff
@@ -49,8 +52,16 @@ type IGMPSnooper struct {
 	ofClient      openflow.Client
 	ifaceStore    interfacestore.InterfaceStore
 	eventCh       chan *mcastGroupEvent
-	queryInterval time.Duration
 	validator     types.MulticastValidator
+	queryInterval time.Duration
+	// igmpReportANPStats is a map that saves AntreaNetworkPolicyStats of IGMP report packets.
+	// The map can be interpreted as
+	// map[UID of the AntreaNetworkPolicy]map[name of AntreaNetworkPolicy rule]statistics of rule
+	igmpReportANPStats      map[apitypes.UID]map[string]*types.RuleMetric
```

Comment:
Use `types.RuleMetric` as the value of the map? Then you  don't need to worry the update on the pointer.

Reply:
updated as @tnqn suggested https://github.com/antrea-io/antrea/pull/3449#discussion_r894246561.